### DISCUSSION TITLE
Video transcoding on demand

### DISCUSSION THREAD

**Discussion by skarekrow** [immich-app/immich#12518] [Q&A] [answered]
Hello! I was just wondering if you guys considered only transcoding on demand in real time when a client tries to view a video that is not compatible with it. I have a lot of videos in my library and having the transcoding feature on is not only expensive in terms of cpu cycles to first transcode these *all* once, but also storage to hold an inferior copy anyways. I see transcodes as a necessary evil and essentially ephemeral for that reason.

A point of comparison is the Memories app for Nextcloud which does this. If you have any hardware transcoding available it makes for a fairly instant start to a video that would otherwise not play with transcoding disabled in Immich for example. As it starts transcoding and streaming the chunks in realtime akin to what Plex/Emby/Jellyfin do.

Thanks for the great project!

**Comment by mertalev** [comment]
This is something we want to support at some point, but it isn't a priority at the moment. There are other advantages besides storage, such as being able to tailor the transcode to the client rather than having a one-size-fits-all transcode.

FWIW, there are also advantages to transcoding ahead of time like this, such as lower latency and higher possible quality due to the lack of a time constraint.

**Comment by bo0tzz** [comment, marked as answer]
#7666

**Reply by skarekrow** [reply]
Sorry for some reason github's search brought up hundreds of irrelevant tickets, thanks for linking the right one!
